src/charm.py

```diff
@@ -43,12 +45,15 @@ def __init__(self, *args):
         # adapter and not directly with the framework.
         self.fw_adapter = FrameworkAdapter(self.framework)
         self.prometheus = PrometheusInterface(self, 'http-api')
+        self.am_rel_name = 'alertmanager'
+        self.am_interface = AlertManagerInterface(self, self.am_rel_name)
         # Bind event handlers to events
         event_handler_bindings = {
             self.on.start: self.on_start,
             self.on.config_changed: self.on_config_changed,
             self.on.upgrade_charm: self.on_upgrade,
             self.on.stop: self.on_stop,
+            self.am_interface.on.new_am_rel: self.on_new_am_rel
```

Comment:
I'd suggest to use `relation` instead of `rel` - just for some code clearness and readability.

Reply:
good point.